### ISSUE TITLE
Wrong date Position 

### ISSUE THREAD

**Issue by Salmankhan033** [hoangnm/react-native-week-view#206]
I pass the date and then sometimes return the wrong date position.
He Header data is correct but the body date position is wrong.
They return  `targetIndex` in negative  value 
and  `initialDates[targetIndex]` is undefined 

```
  goToDate = (targetDate, callBack, animated = true) => {
        console.log("myDate...",targetDate)
        const {initialDates} = this.state
        const {numberOfDays} = this.props

        const currentDate = moment(initialDates[this.currentPageIndex]).startOf('day')
        const deltaDay = moment(targetDate).startOf('day').diff(currentDate, 'day')
        const deltaIndex = Math.floor(deltaDay / numberOfDays)
        const signToTheFuture = this.getSignToTheFuture()
        const targetIndex = this.currentPageIndex + deltaIndex * signToTheFuture
        this.goToPageIndex(targetIndex, animated)
        console.log("goToDate..///",currentDate)
        console.log("targetIndex..///",targetIndex)
        console.log("initialDates[targetIndex]..///",initialDates[targetIndex])
        if (callBack) {
            callBack(moment(initialDates[targetIndex]).toDate())
        }
    }
```
<img width="710" alt="Screenshot 2022-05-23 at 8 37 13 PM" src="https://user-images.githubusercontent.com/58082294/169855931-240597db-d5b0-4c73-8922-4951232f8f23.png">

**Comment by pdpino** [comment]
The `initialDates` array is updated later, so is OK that `initialDates[targetIndex]` is undefined at that moment.

[At the end of the method `goToPageIndex(targetIndex)`](https://github.com/hoangnm/react-native-week-view/blob/master/src/WeekView/WeekView.js#L248) there is a call to `this.setState(...)`, with the updated `initialDates` and `currentMoment`

**Comment by pdpino** [comment]
If you want to use a callback after the scroll, you would probably have to call it at the end of [this `scrollTo()` function](https://github.com/hoangnm/react-native-week-view/blob/master/src/WeekView/WeekView.js#L213), which is inside the `goToPageIndex()` method. I have not tested it though

**Comment by Salmankhan033** [comment]
How to solve it ?

**Comment by Salmankhan033** [comment]
not working.

**Comment by SwikarBhattarai** [comment]
has this issue been solved? please reply

**Comment by pdpino** [comment]
I closed as something different than #185.

This issue is similar, but I did not understand if it was duplicated, the issue was the undefined, the negative value, etc
